src/directives/selection/selection.directive.ts

```diff
@@ -57,10 +58,10 @@ export class SelectionDirective<T> implements AfterContentInit, OnDestroy {
     private _onDestroy = new Subject<void>();
 
     /** Store the previous selection so we don't emit more than we have to */
-    private _lastSelection: ReadonlyArray<T>;
+    private _lastSelection: ReadonlyArray<T> = [];
 
     constructor(private _selectionService: SelectionService<T>, private _cdRef: ChangeDetectorRef) {
-        _selectionService.selection$.pipe(takeUntil(this._onDestroy), debounceTime(0)).subscribe(items => {
+        _selectionService.selection$.pipe(debounceTime(0), takeUntil(this._onDestroy)).subscribe(items => {
```

Comment:
What's the purpose of `debounceTime(0)`, waiting for 0ms? Is it needed for some reason?

Reply:
Yes, so lets say we are in `row` mode, clicking on an item in row mode will select only that one row. What actually occurs is:
- deselect all rows
- select target row

Both of these actions will emit a new value on the observable. However the first is a transitional state, the other is the final state. There are many cases where this may fire multiple times before arriving on the final state.

The key point here that all these events fire synchronously the same tick of the event loop. 
`debounceTime(0)` will only proceed on a subsequent tick of the event loop, at which point all synchronous events that could have happened will have happened.

So, in short, the subscribe function will only get called after all the events have fired and the state is "stable".

Reply:
That makes sense, thanks!